packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -136,6 +138,10 @@ const VirtualScrollerPinnedColumns = styled('div', {
   }),
   ...(ownerState.side === GridPinnedPosition.left && { left: 0, float: 'left' }),
   ...(ownerState.side === GridPinnedPosition.right && { right: 0, float: 'right' }),
+  ...(ownerState.side === GridPinnedPosition.right &&
+    ownerState.showCellRightBorder && {
+      borderLeft: `1px solid ${getBorderColor(theme)}`,
```

Comment:
Good point!

>  If the user wants to change the color of the cell border he will have to change in two places: here and in .MuiDataGrid-withBorder.

But there's more places with border color anyway:
- https://github.com/mui/mui-x/blob/140f39480052c1ca41561eea6fcd28c3a07e1da9/packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts#L66
- https://github.com/mui/mui-x/blob/140f39480052c1ca41561eea6fcd28c3a07e1da9/packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts#L169
- https://github.com/mui/mui-x/blob/140f39480052c1ca41561eea6fcd28c3a07e1da9/packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts#L249

Maybe we should add separate `borderColor` class and apply it everywhere border is used?

Reply:
Ok, I've split `withBorder` styles into two classes:
- `withBorder` that applies `border-color` only
- `withRightBorder` that applies `border-right-width` and `border-right-style` only

I've also used `withBorder` to apply border color to right-pinned container.

Reply:
We miss a `withLeftBorder` applying `border-left-width` and `border-right-style` to the cell, not to the right pinned columns container.

Reply:
There are 2 problems with that:

1. I don't see a way to set `showLeftBorder={true}` only for first cells in the right-pinned section.
2. Applying left border to cell instead of right pinned columns container results in double border:

	  borderLeft on right-pinned columns container
	  <img width="302" alt="Screenshot 2022-04-04 at 19 54 15" src="https://user-images.githubusercontent.com/13808724/161603955-8341afd3-4050-4ba4-baeb-306197e7e1bc.png">
	  borderLeft on cell
	  <img width="303" alt="Screenshot 2022-04-04 at 19 52 52" src="https://user-images.githubusercontent.com/13808724/161603960-970ae665-ea14-487a-b5a9-b0a970264909.png">
	I didn't find a way to avoid this double border.

Reply:
I've renamed `showCellRightBorder` to `showCellVerticalBorder`, `showColumnRightBorder` to `showColumnVerticalBorder` and added `withBorderColor` CSS class that sets the border color for both cells and column headers